src/lib.rs

```diff
@@ -246,9 +399,18 @@ impl SecretKey {
             return None;
         }
         let Ciphertext(ref u, ref v, _) = *ct;
-        let g = u.into_affine().mul(self.0);
+        let g = u.into_affine().mul(*self.0);
```

Comment:
This is something we should also keep in mind for later: We're passing our precious secrets into `pairing`'s arithmetic by value here, which probably won't get zeroed nor mlocked, and possibly will be copied. (Or used in a few reversible computation steps whose outputs won't get zeroed, etc.)

Reply:
Yea, but I don't know what we could do with another library's stack. The `clear-on-drop` crate has a some cool [`clear-stack-on-return`](https://github.com/cesarb/clear_on_drop/blob/master/src/clear_stack_on_return.rs) functionality, but that's only for `FnMut` and `FnOnce`. The curve ed25519 crate put out a release recently where they added some clear-on-drop functionality for heap allocated secrets, but from their [`README`](https://github.com/dalek-cryptography/curve25519-dalek):

> we do not attempt to zero stack data